workflow/controller/operator.go

```diff
@@ -2150,14 +2145,14 @@ func (woc *wfOperationCtx) executeScript(nodeName string, templateScope string,
 		executionDeadline:   opts.executionDeadline,
 	})
 	if err != nil {
-		return woc.checkForbiddenErrorAndResubmitAllowed(err, node.Name, tmpl)
+		return woc.requeueIfTransientErr(err, node.Name)
 	}
 	return node, err
 }
 
-func (woc *wfOperationCtx) checkForbiddenErrorAndResubmitAllowed(err error, nodeName string, tmpl *wfv1.Template) (*wfv1.NodeStatus, error) {
-	if (apierr.IsForbidden(err) || apierr.IsTooManyRequests(err)) && isResubmitAllowed(tmpl) {
-		// Our error was most likely caused by a lack of resources. If pod resubmission is allowed, keep the node pending
+func (woc *wfOperationCtx) requeueIfTransientErr(err error, nodeName string) (*wfv1.NodeStatus, error) {
```

Comment:
This changes the semantics, because we are using `IsTransientErr` now, which itself return true for a different set of errors.